zerver/lib/exceptions.py

```diff
@@ -31,6 +31,7 @@ class ErrorCode(Enum):
     REQUEST_CONFUSING_VAR = auto()
     INVALID_API_KEY = auto()
     INVALID_ZOOM_TOKEN = auto()
+    INVALID_REQUEST = auto()
```

Comment:
Adding this into the middle of the list _changes all of the later IDs_, making them mean something else to clients that rely on them.

You can only add new values to the end of this list.

Please also add a comment to this to clarify that to the next folks that come along.

Reply:
This is an `Enum`, not an `IntEnum`. While there are internal integer values you can get explicitly with `INVALID_REQUEST.value`, that doesn’t seem like something you’d do by accident. Meanwhile, the previous implementation could not be pickled, leading to indecipherable errors from Django’s multiprocessing-based parallel test runner when tests failed.

Reply:
Well, at the very least we clearly regressed in having that invariant be expressed for anyone reading the source code, as evidenced by Alex's comment above. Originally we had:
```py
class AbstractEnum(Enum):
    """An enumeration whose members are used strictly for their names."""
    # …

class ErrorCode(AbstractEnum):
```

and that docstring was lost in 6ae5be6e75e7000dcd1855b378239491646a13c9 without getting replaced.

(FWIW writing `error_code.value` somewhere doesn't seem like that unlikely a mistake to me.)

Reply:
> Meanwhile, the previous implementation could not be pickled, leading to indecipherable errors from Django’s multiprocessing-based parallel test runner when tests failed.

This is a perfect example of the sort of information that would be helpful to include in a commit message.